source/validations.ts

```diff
@@ -0,0 +1,150 @@
+// /source/validations.ts
+// The validation functions
+
+import type { Request } from 'express'
+
+/**
+ * An error thrown/returned when a validation error occurs.
+ */
+class ValidationError extends Error {
+	name: string
+	code: string
+	message: string
```

Comment:
I just tested - we get message automatically by extending Error, so we don't need to define it here.